great_expectations/core/batch_manager.py

```diff
@@ -0,0 +1,166 @@
+import logging
+from collections import OrderedDict
+from typing import Dict, List, Optional
+
+from great_expectations.core.batch import (
+    Batch,
+    BatchData,
+    BatchDefinition,
+    BatchMarkers,
+)
+from great_expectations.core.id_dict import BatchSpec
+
+logger = logging.getLogger(__name__)
+logging.captureWarnings(True)
+
+
+class BatchManager:
+    def __init__(
+        self,
+        execution_engine: "ExecutionEngine",  # noqa: F821
+        batch_list: Optional[List[Batch]] = None,
+    ) -> None:
+        """
+        Args:
+            execution_engine: The ExecutionEngine to be used to access cache of loaded Batch objects.
+            batch_list: List of Batch objects available from external source (default is None).
+        """
+        self._execution_engine = execution_engine
+
+        self._active_batch_id = None
+        self._active_batch_data_id = None
+
+        if batch_list is None:
+            batch_list = []
+
+        self._batch_cache = {}
+        self._batch_data_cache = {}
+
+        self.load_batch_list(batch_list=batch_list)
+        if len(batch_list) > 1:
+            logger.debug(
```

Comment:
Should this be `info` instead of `debug`?

Reply:
@NathanFarmer Fixed (in next commit) -- thanks!